docs/adr/adr-006-befp.md

```diff
@@ -0,0 +1,71 @@
+# ADR #006: Celestia-Node Bad encoding fraud proof(BEFP)
+
+## Changelog
+
+- 2021.03.03 - init commit
+
+## Authors
+
+@vgonkivs @Bidon15 @adlerjohn
+
+## Context
+
+In case when node receives errors `ErrByzantineRow`/`ErrByzantineCol` it should generate a fraud proof that broadcasts message to light clients and informs that block could be malicious.
+
+## Decision
+
+Started disscussion within:
+
+https://github.com/celestiaorg/celestia-node/issues/4
+
+https://github.com/celestiaorg/celestia-node/issues/263
+
+## Detailed Design
+It should be generated when after repairing the entire block, we detect recovered data does not match with its respective row/column roots. The result of RepairExtendedDataSquare will be an error ErrByzantineRow/ErrByzantineCol:
+
+```go
+// ErrByzantineRow is thrown when a repaired row does not match the expected row Merkle root.
+type ErrByzantineRow struct {
+   RowNumber uint     // Row index
+   Shares    [][]byte // Pre-repaired row shares. Missing shares are nil.
+}
+```
+
+```go
+// ErrByzantineCol is thrown when a repaired column does not match the expected column Merkle root.
+type ErrByzantineCol struct {
+   ColNumber uint     // Column index
+   Shares    [][]byte // Pre-repaired column shares. Missing shares are nil.
+}
+```
```

Comment:
I do not think we need to list both of them here. ErrByzantineRow is sufficient and then just add a note: "and for columns, respectively."